crates/types/src/traits/block_contents.rs

```diff
@@ -23,6 +23,12 @@ use crate::{
     vid::{vid_scheme, VidCommitment, VidSchemeType},
 };
 
+/// Trait for structures that need to be unambiguously encoded as bytes.
+pub trait EncodeBytes {
+    /// Encode `&self`
+    fn encode(&self) -> Arc<[u8]>;
```

Comment:
What's the rationale for `Arc<[u8]>` vs `impl Iterator<Item = u8>`?

Reply:
That's what `BlockPayload::encode` returns, and my first instinct was to be consistent if we want to replace `BlockPayload::encode` with `BlockPayload: EncodeBytes`
Rationale for that originally probably was that we often pass the result around between tasks, therefore `Arc`.
I think Iterator would be unwieldy, in most if not all of cases we don't need to iterate over it, so we'd just end up with `collect`s all over the place for no clear payoff.
That said, we totally could leave `Arc`ing to the caller and make it just return a `Vec` or even make an `encode_into(&mut Vec)` and `encode` a wrapper around that

Reply:
Thanks for explaining, I think keeping it the way it is fine